exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/Entity.kt

```diff
@@ -125,15 +125,15 @@ open class Entity<ID : Comparable<ID>>(val id: EntityID<ID>) {
                     }
                 }
                 else -> {
-                    // @formatter:off
+                    val castReferee = reference.referee<REF>()!!
+                    val baseReferee = (castReferee.columnType as? EntityIDColumnType<REF>)?.idColumn ?: castReferee
                     factory.findWithCacheCondition({
                         reference.referee!!.getValue(this, desc) == refValue
                     }) {
-                        reference.referee<REF>()!! eq refValue
+                        baseReferee eq refValue
```

Comment:
This forces the query clause to use the underlying/wrapped type of `EntityIDColumnType` when setting statement parameters.

The elvis op is necessary in case the cast fails, which covers cases when an entity may have a foreign key on a target refColumn that is not actually an entityID column.